crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1302,15 +1302,6 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
     ) -> impl Iterator<Item = &'a Requirement> {
         self.overrides
             .apply(dependencies)
-            .flat_map(|requirement| {
-                iter::once(requirement).chain(
-                    // If the requirement was constrained, add those constraints.
-                    self.constraints
-                        .get(&requirement.name)
-                        .into_iter()
-                        .flatten(),
-                )
-            })
```

Comment:
The key is that we shouldn't apply the constraints here; we only apply them _after_ we've filtered requirements.